test/cql-pytest/test_udf.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+# Copyright 2021-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+#############################################################################
+# Tests for user-defined functions (UDF)
+#############################################################################
+
+import pytest
+from util import unique_name, new_function
+from cassandra.protocol import InvalidRequest
+
+# Unfortunately, while ScyllaDB and and Cassandra support the same UDF
+# feature, each supports a different choice of languages. In particular,
+# Cassandra supports Java and ScyllaDB doesn't - but ScyllaDB supports
+# Lua. The following fixture can be used to check if the server supports
+# Java UDFs - and if it doesn't, the test should use Lua.
+@pytest.fixture(scope="session")
+def has_java_udf(cql, test_keyspace):
+    try:
+        with new_function(cql, test_keyspace, "(i int) CALLED ON NULL INPUT RETURNS int LANGUAGE java AS 'return 42;'"):
+            return True
+    except InvalidRequest as e:
+        # When the language "java" is not supported, Scylla prints the
+        # error message "Language 'java' is not supported". Cassandra
+        # isn't expected to fail here because it does support Java.
+        # Any other error here indicates an unexpected bug.
+        assert "Language 'java' is not supported" in str(e)
+        return False
+
+# Test that if we have two functions with the same name, just with
+# different parameters, DROP FUNCTION requires disambiguating which
+# function you want to drop. Without parameters, the DROP will be an
+# error. But when there is only one function with the same name, it's
+# not necessary to give the parameters in the DROP command.
+def test_drop_udf_with_same_name(cql, test_keyspace, has_java_udf):
+    if has_java_udf:
+        body1 = "(i int) CALLED ON NULL INPUT RETURNS bigint LANGUAGE java AS 'return 42L;'"
```

Comment:
I see the strings defining functions repeated a couple of times. I wonder if it wouldn't be better to put a function definition into an f-string and parameterize with parameters a specific function takes. That at least should prevent us from accidentally creating not-overloaded functions, if not cleaning the code *a bit* (or maybe it'd decrease the readability). That's just an idea, I leave it up to you.

Reply:
You're right that in this silly example, the four bodies are the same, more or less. I'll do this.

Reply:
I'll need to switch from bigint to int to make them exactly the same ;-)

Reply:
I changed my mind - all this cleverness doesn't make the test shorter, and makes it harder to understand than just seeing these 4 lines inside clear ifs.
By the way, please note we can't "accidentally create non-overloaded functions" - the name of the function - `fun` - is defined below. It's not part of the body.

Reply:
OK, thanks for trying anyways.